python/instrumentation/openinference-instrumentation-vertexai/src/openinference/instrumentation/vertexai/_accumulator.py

```diff
@@ -0,0 +1,109 @@
+from collections import defaultdict
+from copy import deepcopy
+from typing import (
+    Any,
+    Callable,
+    DefaultDict,
+    Dict,
+    Iterable,
+    Iterator,
+    List,
+    Mapping,
+    Optional,
+    Tuple,
+)
+
+__all__ = (
+    "_KeyValuesAccumulator",
+    "_StringAccumulator",
+    "_IndexedAccumulator",
+)
+
+
+class _KeyValuesAccumulator:
```

Comment:
These accumulators are copied from the `openai` instrumentor. We can move them to `instrumentation` package once they are sufficiently tested and are stable.